packages/dds/tree/src/feature-libraries/defaultFieldKinds.ts

```diff
@@ -667,16 +661,10 @@ export const optional: FieldKind<OptionalFieldEditor> = new FieldKind(
 		encoder: optionalFieldEncoder,
 		editor: optionalFieldEditor,
 
-		intoDelta: (change: OptionalChangeset, deltaFromChild: ToDelta, reviver: NodeReviver) => {
+		intoDelta: (change: OptionalChangeset, deltaFromChild: ToDelta) => {
 			const update = change.fieldChange?.newContent;
-			let content: JsonableTree | ITreeCursorSynchronous | undefined;
-			if (update === undefined || "set" in update) {
-				content = update?.set;
-			} else {
-				const revision = update.revert;
-				assert(revision !== undefined, 0x478 /* Unable to revert to undefined revision */);
-				content = reviver(revision, 0, 1)[0];
-			}
+			const content: JsonableTree | ITreeCursorSynchronous | undefined =
+				update === undefined || "set" in update ? update?.set : update.revert;
 			const insertDelta = deltaFromInsertAndChange(
 				content,
```

Comment:
This function doesn't handle the case where `content` is `ITreeCursorSynchronous`. The reason the type system doesn't complain is that pretty much anything can be a `JsonableTree`.

Before you fix this, it would be good to add the test that would have detected the issue. In fact I'm surprised the existing test didn't detect that. It might need tweaking.